src/main/java/wedlog/address/logic/commands/GuestEditCommand.java

```diff
@@ -0,0 +1,329 @@
+            Email updatedEmail = isEmailEdited ? email : guestToEdit.getEmail().orElse(null);
+            Address updatedAddress = isAddressEdited ? address : guestToEdit.getAddress().orElse(null);
+            RsvpStatus updatedRsvp = isRsvpEdited ? rsvp : guestToEdit.getRsvpStatus();
+            DietaryRequirements updatedDietary = isDietaryEdited ? dietary : guestToEdit.getDietaryRequirements();
+            TableNumber updatedTable = isTableEdited ? table : guestToEdit.getTableNumber().orElse(null);
+            Set<Tag> updatedTags = isTagsEdited ? Collections.unmodifiableSet(tags) : guestToEdit.getTags();
+
+            return new Guest(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedRsvp, updatedDietary,
+                    updatedTable, updatedTags);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof EditGuestDescriptor)) {
+                return false;
+            }
+
+            EditGuestDescriptor otherEditGuestDescriptor = (EditGuestDescriptor) other;
+            return Objects.equals(name, otherEditGuestDescriptor.name)
+                    && Objects.equals(phone, otherEditGuestDescriptor.phone)
+                    && Objects.equals(email, otherEditGuestDescriptor.email)
+                    && Objects.equals(address, otherEditGuestDescriptor.address)
+                    && Objects.equals(rsvp, otherEditGuestDescriptor.rsvp)
+                    && Objects.equals(dietary, otherEditGuestDescriptor.dietary)
+                    && Objects.equals(table, otherEditGuestDescriptor.table)
+                    && Objects.equals(tags, otherEditGuestDescriptor.tags)
+                    && isNameEdited == otherEditGuestDescriptor.isNameEdited
+                    && isPhoneEdited == otherEditGuestDescriptor.isPhoneEdited
+                    && isEmailEdited == otherEditGuestDescriptor.isEmailEdited
+                    && isAddressEdited == otherEditGuestDescriptor.isAddressEdited
+                    && isRsvpEdited == otherEditGuestDescriptor.isRsvpEdited
+                    && isDietaryEdited == otherEditGuestDescriptor.isDietaryEdited
+                    && isTableEdited == otherEditGuestDescriptor.isTableEdited
+                    && isTagsEdited == otherEditGuestDescriptor.isTagsEdited;
+        }
+
+        @Override
+        public String toString() {
+            return new ToStringBuilder(this)
+                    .add("name", name)
+                    .add("phone", phone)
+                    .add("email", email)
+                    .add("address", address)
+                    .add("rsvp status", rsvp)
+                    .add("dietary requirement", dietary)
+                    .add("table number", table)
+                    .add("tags", tags)
+                    .toString();
+        }
```

Comment:
Since this method should encapsulate the descriptor, perhaps the string should contain whether the field is edited or not?
```suggestion
        @Override
        public String toString() {
            return new ToStringBuilder(this)
                    .add("name", name)
                    .add("phone", phone)
                    .add("email", email)
                    .add("address", address)
                    .add("rsvp status", rsvp)
                    .add("dietary requirement", dietary)
                    .add("table number", table)
                    .add("tags", tags)
                    .add("isNameEdited", isNameEdited)
                    .add("isPhoneEdited", isPhoneEdited)
                    .add("isEmailEdited", isEmailEdited)
                    .add("isAddressEdited", isAddressEdited)
                    .add("isRsvpEdited", isRsvpEdited)
                    .add("isDietaryEdited", isDietaryEdited)
                    .add("isTableEdited", isTableEdited)
                    .add("isTagsEdited", isTagsEdited)
                    .toString();
        }
```